shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/utility/DateKMM.kt

```diff
@@ -0,0 +1,18 @@
+package com.kcc.kmmhackathon.shared.utility
+
+expect class DateKMM {
+    constructor()
+
+    constructor(time: Long)
+
+    /**
+     * Compare this instance with the other instance.
+     *
+     * @param other: Another date instance.
+     * @return 0 if the other is equal to this instance; a value less than 0 if this instance is before the other;
+     *          and a value greater than 0 if this instance is after the other.
+     */
+    fun compareTo(other: DateKMM): Int
```

Comment:
This is the documentation for the Java data compareTo function.
It's not the easiest function to understand, we could add our own compare function that returns a bool but I think then it would take a custom implementation of Date on Android side - but happy to do it if everyone thinks this is a bit gross

Reply:
I think there are some comparison APIs on the iOS side that return an enum (basically an Int) so I think it's okay to leave, might be nice to wrap up into something a bit more Swifty but fine as-is too :)

Reply:
I extended ShareDate to have `isBefore`, `isAfter` and `isEqual` functions to make this a bit easier